src/Kafka/Consumer/Subscription.hs

```diff
@@ -17,6 +17,18 @@ import           Data.Text            (Text)
 import           Kafka.Consumer.Types (OffsetReset (..))
 import           Kafka.Types          (TopicName (..))
 
+-- | A consumer subscription to a topic.
+--
+-- ==== __Examples__
+--
+-- Typically you don't call the constructor directly, but combine settings:
+--
+-- @
+-- consumerSub :: 'Subscription'
+-- consumerSub = 'topics' ['TopicName' "kafka-client-example-topic"]
+--         <> 'offsetReset' 'Earliest'
+--         <> 'extraSubscriptionProps' (fromList [("prop1", "value 1"), ("prop2", "value 2")])
```

Comment:
I added a use of `extraSubscriptionProps` but I wasn't sure which other properties were configurable here vs to be configured in `ConsumerProperties`